resources/rest-service/cloudify/migrations/versions/9aa6f74c9653_4_0_1.py

```diff
@@ -397,6 +397,7 @@ def upgrade():
 
 def downgrade():
     # ### commands auto generated by Alembic - please adjust! ###
+    op.execute('LOCK TABLE users IN ACCESS EXCLUSIVE MODE')
```

Comment:
The issue is that during the test, while the migration runs and tries to grab all kinds of locks on the users table (because of adding FK's etc), the test framework (or the CLI, actually) is going to hammer the server by polling the execution status - which runs the auth logic - which updates `last_login_at` - which needs to grab a writer lock on the user row.

This was causing a deadlock in cases when the `last_login_at` update happened somewhere in the middle of this downgrade transaction block.

Btw, to easier reproduce this deadlock (which seems to be very dependent on the environment, and luck - like most race conditions), one way is to edit the test framework code that polls execution status (in `wait_for_execution_to_end`), to poll eg. every 10 msec, not 500.

This change just grabs a lock on the whole table, so all requests will simply have to wait for the downgrade to finish, but should fix the deadlocks.